### PR TITLE
[swift-test] inherit enviroment when running xctest

### PR DESCRIPTION
**Pull request by aciidb0mb3r**
Fixes https://bugs.swift.org/browse/SR-1396

### CODE REVIEW

File: Sources/swift-test/test.swift [hunk 1]
```diff
@@ -31,7 +32,7 @@ func test(path: String, xctestArg: String? = nil) throws -> Bool {
     }
 #endif
 
-    let result: Void? = try? system(args)
+    let result: Void? = try? system(args, environment: NSProcessInfo.processInfo().environment)
```
Comment (Bouke) [inline Sources/swift-test/test.swift]: That would be the same as passing no environment (nil) to either posix_spawn or NSTask.environment, wouldn't that be cleaner?

Comment (aciidb0mb3r) [inline Sources/swift-test/test.swift]: I don't think posix_spawn() inherits env if you pass nil.

Comment (ddunbar) [inline Sources/swift-test/test.swift]: Can we get a comment here to clarify the intent, so it is obvious why we pass the full env?

Comment (aciidb0mb3r) [inline Sources/swift-test/test.swift]: done

**Comment by ddunbar** [comment]
@swift-ci please test and merge

**Comment by aciidb0mb3r** [comment]
@swift-ci please test and merge

**Pull request merged**